### DISCUSSION TITLE
Global matplotlib style with objects interface

### DISCUSSION THREAD

**Discussion by juanitorduz** [mwaskom/seaborn#3005] [Q&A] [answered]
Hey! thanks for the amazing 0.12 release 🚀 !

I want to use the new object interface while using a global `matplotlib` style following https://seaborn.pydata.org/generated/seaborn.objects.Plot.theme.html#seaborn.objects.Plot.theme

I have tried:

```python
import matplotlib.pyplot as plt
import seaborn as sns
import seaborn.objects as so

anscombe = sns.load_dataset("anscombe")

plt.style.use("bmh")

sns.set_theme({**style.library["bmh"]})
sns.set_style({**style.library["bmh"]})

sns.set_theme(rc=style.library["bmh"])
sns.set_style(rc=style.library["bmh"])


p = (
    so.Plot(anscombe, "x", "y")
    .facet("dataset", wrap=2)
    .add(so.Line(), so.PolyFit(order=1))
    .add(so.Dot())
)
p
```

but I am still getting:

![image](https://user-images.githubusercontent.com/22996444/188645354-4186be5e-6872-4fe1-bc5b-b2625804454e.png)

Any tips on how to make this work? Thanks!

**Remark:** With the "old" seaborn's functional API it works as expected.

**Comment by mwaskom** [comment]
There is some context in the PR: https://github.com/mwaskom/seaborn/pull/2929

One of the design principles in the `Plot` interface is to avoid all of the magical global state stuff that matplotlib (and the function interface) does. So, no pyplot, no global rcParams, etc. I do expect that there will be some ability to set default behavior through `Plot` itself, but I'm not sure exactly what that should look like yet. So for now you'll need to set the them on every plot (sorry), like this

```python
(
    so.Plot(anscombe, "x", "y")
    .facet("dataset", wrap=2)
    .add(so.Line(), so.PolyFit(order=1))
    .add(so.Dot())
    .theme(plt.style.library["bmh"])
)
```
![image](https://user-images.githubusercontent.com/315810/188752394-eb9f0a2e-9e33-4378-9258-5462be4ebc84.png)

**Comment by juanitorduz** [comment]
Thank you very much for the explanation! 😊 I'll keep exploring the new object API 🚀

**Comment by ravwojdyla** [comment]
FYI in my case to "propagate" the global style changes:

```py
import seaborn as sns
import seaborn.objects as so
import matplotlib.pyplot as plt

my_palette = ...

sns.set_theme(style="whitegrid", palette=my_palette)
plt.rc("figure", figsize=(16, 4), dpi=150)
```

To the Plot API, I had to:

```py
(
    so.Plot(...)
    .scale(color=sns.color_palette())
    .theme(plt.rcParams)
)
```

I'm not entirely sure that's optimal/best solution, would appreciate input if there's a better way. Also looking forward to defaults in Objects API 🤞

**Reply by mwaskom** [reply]
AFK but wouldn’t think you need both there; what is in the dict returned by `axes_style()` that isn’t in `rcParams`?

**Reply by ravwojdyla** [reply]
@mwaskom ah, right `theme(plt.rcParams)`  is enough - thanks. Updated the comment.

**Comment by mwaskom** [comment, marked as answer]
See #3223
